amazon-s3-quickstart/README.md

```diff
@@ -0,0 +1,99 @@
+
+S3 listens on `localhost:8008` for REST endpoints.
+
+Create an AWS profile for your local instance using AWS CLI:
+
+```
+$ aws configure --profile localstack
+AWS Access Key ID [None]: test-key
+AWS Secret Access Key [None]: test-secret
+Default region name [None]: us-east-1
+Default output format [None]:
+```
+
+## Create bucket
+
+Create a S3 bucket using AWS CLI and the localstack profile.
+`aws s3 mb s3://quarkus.s3.quickstart --profile localstack --endpoint-url=http://localhost:8008`
+
+# Run the demo in dev mode
+
+- Run `./mvnw clean package` and then `java -jar ./target/amazon-s3-quickstart-1.0-SNAPSHOT-runner.jar`
+- In dev mode `./mvnw clean quarkus:dev`
+
+Go to `http://localhost:8080/s3.html`, it should show a simple App to manage files on your Bucket. 
+You can upload files to the bucket via the form.
+
+Alternatively, go to `http://localhost:8080/async-s3.html` with the simple App communicating with Async resources.
+
+# Running in native
+
+You can compile the application into a native executable using:
+
+`./mvnw clean package -Pnative`
+
+and run with:
+
+`./target/amazon-s3-quickstart-1.0-SNAPSHOT-runner` 
+
+
+# Running native in container (using localstack)
+
+Build a native image in container by running:
+
+`./mvnw package -Pnative -Dnative-image.docker-build=true`
+
+Build a docker image:
+`docker build -f src/main/docker/Dockerfile.native -t quarkus/amazon-s3-quickstart .`
+
+Create a network that connect your container with localstack
+`docker network create localstack`
+
+Start localstack and connect to the network
+`docker run --rm --network=localstack --name localstack -p 8008:4572 -e SERVICES=s3 -e START_WEB=0 -d localstack/localstack`
+
+Don't forget to create a S3 bucket (see above). And run quickstart container connected to that network (note that we're using internal port of the S3 localstack)
+`docker run -i --rm --network=localstack -p 8080:8080 quarkus/amazon-s3-quickstart -Dquarkus.s3.endpoint-override=http://localstack:4572`
+
+Go to `http://localhost:8080/s3.html` or `http://localhost:8080/async-s3.html`
+
+# Using AWS account
+
+Before you can use the AWS SDKs with S3, you must get an AWS access key ID and secret access key. 
+For more information, see:
+ - [Sign up for AWS and Create an IAM User](https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/signup-create-iam-user.html)
+ - [Set Up AWS Credentials and Region for Development](https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/setup-credentials.html)
+
+Create a S3 bucket using AWS CLI using your default AWS profile
+`aws s3 mb s3://quarkus.s3.quickstart.11.22.33`
+
+**NOTE: Please assure the bucket name you created is unique across AWS S3. 
+See [Amazon S3 Bucket Naming Requirements](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules)**
+
+## Run demo
+
+You can run the demo the same way as for a local instance, but you need to change the `application.properties` as follows:
```

Comment:
Possible to add a note to comment out the `quarkus.s3.endpoint-override` property in the config file?  

I followed the instructions and then had to fight with it a little to work out why it wasn't working, the command line version sets that to = '' yet it does not override what's in the properties file if you forget to do it.